src/dialog-editor/components/abstractModal.ts

```diff
@@ -0,0 +1,36 @@
+import * as _ from 'lodash';
+
+class ModalController {
+  private uibModalInstance: any;
+  private saveModal: any;
+
+  /*@ngInject*/
+  constructor(private DialogEditor: any) {
+  }
+
+  public closeModal(save: boolean) {
+    if (save) {
+      this.saveModal();
+    }
+    this.uibModalInstance.close();
+  }
+}
+
+export default class AbstractModal {
+  public controller = ModalController;
+  public controllerAs: string = 'vm';
+  public bindings: any = {
+    modalData: '=',
+    elementInfo: '<',
+    categories: '=?',
+    addEntry: '=?',
+    removeEntry: '=?',
+    currentCategoryEntries: '=?',
+    resolveCategories: '=?',
+    modalTabIsSet: '<',
+    modalTabSet: '<',
+    modalTab: '=',
```

Comment:
Some of these bindings are specific to the dialog editor and so the name `AbstractModal` doesn't really make sense.